lib/propolis/src/enlightenment/hyperv/overlay.rs

```diff
@@ -175,7 +207,24 @@ pub(super) struct OverlayPage {
     pfn: Pfn,
 }
 
+impl std::fmt::Debug for OverlayPage {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("OverlayPage")
+            .field("kind", &self.kind)
+            .field("pfn", &self.pfn)
+            .finish()
```

Comment:
take it or leave it: a technique I like for hand-implemented `Debug` impls is to exhaustively destructure `self`, explicitly skipping any fields we don't want to format, like this:
```suggestion
        let Self { kind, pfn, manager: _ } = self;
        f.debug_struct("OverlayPage")
            .field("kind", kind)
            .field("pfn", pfn)
            .finish()
```
This way, if a new field is added, the exhaustive destructuring will fail to compile, serving as a nice little reminder to either add the new field to the `Debug` output or explicitly decide to ignore it.

Reply:
Good call, thanks! I'm also generally very enthusiastic about using exhaustive destructuring for this sort of thing, but it didn't occur to me to do it here. Fixed in 2d23e4e.